tests/collections/tts/test_spectrogram_enhancer.py

```diff
@@ -0,0 +1,186 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+
+import pytest
+import torch
+from einops import rearrange
+from omegaconf import DictConfig
+
+from nemo.collections.tts.models import FastPitchModel, SpectrogramEnhancerModel
+
+
+@pytest.fixture
+def enhancer_config():
+    config = {
+        "model": {
+            "lr": 2e-4,
+            "n_bands": 80,
+            "latent_dim": 192,
+            "style_depth": 4,
+            "network_capacity": 16,
+            "mixed_prob": 0.9,
+            "fmap_max": 192,
+            "spectrogram_min_value": -13.18,
+            "spectrogram_max_value": 4.78,
+            "consistency_loss_weight": 10.0,
+            "gradient_penalty_loss_weight": 10.0,
+            "gradient_penalty_loss_every_n_steps": 4,
+            "spectrogram_predictor_path": None,
+        },
+        "generator_opt": {"_target_": "torch.optim.Adam", "lr": 2e-4, "betas": [0.5, 0.9]},
+        "discriminator_opt": {"_target_": "torch.optim.Adam", "lr": 2e-4, "betas": [0.5, 0.9]},
+    }
+
+    return DictConfig(config)
+
+
+@pytest.fixture
+def enhancer_config_with_fastpitch(fastpitch_model_path, test_data_dir):
+    test_data_dir = Path(test_data_dir)
+    config = {
+        "model": {
+            "lr": 2e-4,
+            "n_bands": 80,
+            "latent_dim": 192,
+            "style_depth": 4,
+            "network_capacity": 16,
+            "mixed_prob": 0.9,
+            "fmap_max": 192,
+            "spectrogram_min_value": -13.18,
+            "spectrogram_max_value": 4.78,
+            "consistency_loss_weight": 10.0,
+            "gradient_penalty_loss_weight": 10.0,
+            "gradient_penalty_loss_every_n_steps": 4,
+            "spectrogram_model_path": fastpitch_model_path,
+            "train_ds": {
+                "dataset": {
+                    "manifest_filepath": str(test_data_dir / "tts/mini_ljspeech/manifest.json"),
+                    "sup_data_path": str(test_data_dir / "tts/mini_ljspeech/sup"),
+                },
+                "dataloader_params": {"batch_size": 3},
+            },
+        },
+        "generator_opt": {"_target_": "torch.optim.Adam", "lr": 2e-4, "betas": [0.5, 0.9]},
+        "discriminator_opt": {"_target_": "torch.optim.Adam", "lr": 2e-4, "betas": [0.5, 0.9]},
+    }
+
+    return DictConfig(config)
+
+
+@pytest.fixture
+def fastpitch_model():
+    model = FastPitchModel.from_pretrained(model_name="tts_en_fastpitch")
+    return model
```

Comment:
I wonder if this should be in a separate test file since this does just test FastPitch NGC checkpoint loading in general.

Reply:
This not a test, this is a fixture, a fixture is not supposed to test something by itself

Is designing reusable testing code something worth of effort? I've been taught e.g. copying-pasting tests is an OK thing to do